mypy/plugins/dataclasses.py

```diff
@@ -460,7 +460,11 @@ def _collect_field_args(expr: Expression) -> Tuple[bool, Dict[str, Expression]]:
         # field() only takes keyword arguments.
         args = {}
         for name, arg in zip(expr.arg_names, expr.args):
-            assert name is not None
+            if name is None:
+                # This means that `field` is used with `**` unpacking,
+                # the best we can do for now is not to fail.
+                # TODO: we can infer what's inside `**` and try to collect it.
```

Comment:
Should we give an error instead? The **kwargs may affect things like `init=False`, which would affect type checking.

Reply:
This is tricky. It is technically not an error and it will work correctly during runtime (if the `kwargs` are correct). It is more like a "temporary" notification / limitation from us.

I have a slight preference for non-raising one.

But, I can make this an explicit error if you wish.

Reply:
Yes, I think that's more how mypy generally behaves. I'm pretty sure you'll get a similar error when using say `NamedTuple(**kwargs)`.

Reply:
Sure! Will do today.